### PR TITLE
Add variable noto fonts

### PR DESCRIPTION
**Pull request by sharpenedblade**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
Livecheck is throwing an error, but the version is actually up to date.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-fonts/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
It says the repo is not popular enough. This font (from the same source repo) is already packaged here as a non-variable font so I dont think this applies here. The github repo is just a download link, the actual font is much more popular.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by bevanjkay** [comment]
Thanks for the submission @sharpenedblade. Is there any reason these versions should not be distributed as the current `font-noto-sans` etc...?

**Comment by sharpenedblade** [comment]
> Thanks for the submission @sharpenedblade. Is there any reason these versions should not be used as the current `font-noto-sans` etc...?

I dont think so, but some older apps dont handle variable fonts right, and I noticed that other fonts were packaged in the same way.

**Comment by bevanjkay** [comment]
If this is the current official distribution of the fonts then yes. 

In other cases it is likely because the variable fonts are packaged separately upstream, so cannot be installed from the same archive (therefore requiring a separate cask).

**Comment by sharpenedblade** [comment]
> In other cases it is likely because the variable fonts are packaged separately upstream, so cannot be installed from the same archive (therefore requiring a separate cask).

Having both regular and variable fonts installed causes inconsistant rendering in a lot of apps.

**Comment by bevanjkay** [comment]
According to the documentation, we prefer variable fonts over static fonts, so this should be ok, unless it causes issues in up to date applications.
https://github.com/Homebrew/homebrew-cask-fonts/blob/master/CONTRIBUTING.md#multiple-font-formats

**Comment by sharpenedblade** [comment]
> According to the documentation, we prefer variable fonts over static fonts, so this should be ok, unless it causes issues in up to date applications. https://github.com/Homebrew/homebrew-cask-fonts/blob/master/CONTRIBUTING.md#multiple-font-formats

I updated the PR

**Comment by bevanjkay** [comment]
Tests are currently failing. But in principle I'm happy with the changes.

**Comment by sharpenedblade** [comment]
> Tests are currently failing. But in principle I'm happy with the changes.

Fixed, the only error is about popularity but there is nothing I can do about that.

File: Casks/font-noto-sans-mono.rb [hunk 1]
```diff
@@ -1,48 +1,14 @@
 cask "font-noto-sans-mono" do
-  version :latest
+  version "2.014"
   sha256 :no_check
```
Comment (bevanjkay) [inline Casks/font-noto-sans-mono.rb:3]: Now that we are using a versioned `url` a checksum can be added to each cask

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/font-noto-sans-mono.rb (hunk 1 above)
Comment (bevanjkay) [reply Casks/font-noto-sans-mono.rb:3]: Should be good to go after that change

Comment (sharpenedblade) [reply Casks/font-noto-sans-mono.rb]: I added checksums

**Comment by sharpenedblade** [comment]
All tests are passing now

File: Casks/font-noto-sans-mono.rb [hunk 2]
```diff
@@ -1,48 +1,19 @@
...
-  font "NotoSansMono-SemiCondensedBold.ttf"
-  font "NotoSansMono-SemiCondensedExtraBold.ttf"
-  font "NotoSansMono-SemiCondensedExtraLight.ttf"
-  font "NotoSansMono-SemiCondensedLight.ttf"
-  font "NotoSansMono-SemiCondensedMedium.ttf"
-  font "NotoSansMono-SemiCondensedSemiBold.ttf"
-  font "NotoSansMono-SemiCondensedThin.ttf"
-  font "NotoSansMono-Thin.ttf"
+  livecheck do
+    url "https://github.com/notofonts/latin-greek-cyrillic.git"
```
Comment (bevanjkay) [inline Casks/font-noto-sans-mono.rb:12]: all of these should be adjusted to

```suggestion
    url :url
```

Sorry for the back and forth @sharpenedblade

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/font-noto-sans-mono.rb (hunk 2 above)
Comment (sharpenedblade) [reply Casks/font-noto-sans-mono.rb]: Fixed

**Review by bevanjkay — APPROVED** [review]

**Comment by bevanjkay** [comment]
Thank you @sharpenedblade

**Pull request merged**